python/cells.cpp

```diff
@@ -0,0 +1,226 @@
+        .def(pybind11::init<>(
+            [](const poisson_schedule_shim& sched, double ratio){
+                return arb::benchmark_cell{sched.schedule(), ratio};}),
+            "schedule"_a, "realtime_ratio"_a)
+        .def("__repr__", [](const arb::spike_source_cell&){return "<benchmark_cell>";})
+        .def("__str__",  [](const arb::spike_source_cell&){return "<benchmark_cell>";});
+
+    pybind11::class_<arb::lif_cell> lif_cell(m, "lif_cell",
+        "A benchmarking cell, used by Arbor developers to test communication performance.");
+
+    lif_cell
+        .def(pybind11::init<>())
+        .def_readwrite("tau_m", &arb::lif_cell::tau_m,  "Membrane potential decaying constant [ms].")
+        .def_readwrite("V_th",  &arb::lif_cell::V_th,   "Firing threshold [mV].")
+        .def_readwrite("C_m",   &arb::lif_cell::C_m,    "Membrane capacitance [pF].")
+        .def_readwrite("E_L",   &arb::lif_cell::E_L,    "Resting potential [mV].")
+        .def_readwrite("V_m",   &arb::lif_cell::V_m,    "Initial value of the Membrane potential [mV].")
+        .def_readwrite("t_ref", &arb::lif_cell::t_ref,  "Refractory period [ms].")
+        .def_readwrite("V_reset", &arb::lif_cell::V_reset, "Reset potential [mV].")
+        .def("__repr__", [](const arb::lif_cell&){return "<lif_cell>";})
+        .def("__str__",  [](const arb::lif_cell&){return "<lif_cell>";});
+
+    pybind11::class_<cell_parameters> cell_params(m, "cell_parameters", "Parameters used to generate the random cell morphologies.");
+    cell_params
+        .def(pybind11::init<>())
+        .def_readwrite("depth", &cell_parameters::max_depth,"The maximum depth of the branch structure.")
+        .def_readwrite("lengths",   &cell_parameters::lengths,  "Length of branch in μm [range].")
+        .def_readwrite("synapses",  &cell_parameters::synapses, "The number of randomly generated synapses on the cell.")
+        .def_readwrite("branch_probs", &cell_parameters::branch_probs, "Probability of a branch occuring [range].")
+        .def_readwrite("compartments", &cell_parameters::compartments, "Compartment count on a branch [range].")
+        .def("__repr__", util::to_string<cell_parameters>)
+        .def("__str__",  util::to_string<cell_parameters>);
+
+    // Wrap cable cell description type.
+    // Cable cells are going to be replaced with a saner API, so we don't go
+    // adding much in the way of interface. Instead we just provide a helper
+    // that will generate random cell morphologies for benchmarking.
+    pybind11::class_<arb::cable_cell> cablecell(m, "cable_cell");
+    m.def("branch_cell", &branch_cell,
+        "Construct a branching cell with a random morphology and synapse end points locations described by params.\n"
+        "gid is an integral value used to seed the random number generator. The unique global identifier of a cell is a good value to use.",
+        "gid"_a,
+        "params"_a=cell_parameters());
+}
```

Comment:
- Shouldn't we name the type `cable_cell` instead of `cablecell`?

Reply:
yes, we should.

Reply:
Though this makes no difference: `cablecell` is just the name of the local variable used in this C++ function: the name in Python is `cable_cell`:
```C++
pybind11::class_<arb::cable_cell> cablecell(m, "cable_cell"); 
                                                                                       ^ name in Python
```

Reply:
It shouldn't, just for consistency again :)